controllers/inferecenceservice_serviceaccount.go

```diff
@@ -63,17 +59,17 @@ func createDelegateClusterRoleBinding(serviceAccountName string, serviceAccountN
 	}
 }
 
-func (r *OpenshiftInferenceServiceReconciler) reconcileSA(inferenceService *inferenceservicev1.InferenceService, ctx context.Context, newSA func(service *inferenceservicev1.InferenceService) *corev1.ServiceAccount) error {
+func (r *OpenshiftInferenceServiceReconciler) reconcileSA(inferenceService *inferenceservicev1.InferenceService, ctx context.Context, serviceAccountName string, serviceAccountNamespace string, newSA func(serviceAccountName string, serviceAccountNamespace string) *corev1.ServiceAccount) error {
 
 	// Initialize logger format
 	log := r.Log.WithValues("inferenceservice", inferenceService.Name, "namespace", inferenceService.Namespace)
 
-	desiredSA := newSA(inferenceService)
+	desiredSA := newSA(serviceAccountName, serviceAccountNamespace)
```

Comment:
Should we delete the old SA, if there is a new name?

Reply:
I have same doubt in this area as well. According to kserve/modelmesh-controller codebase, `ServiceAccount` should be maintained by the user. At the last user just provide the name reference of `ServiceAccount` which he want to link with ServingRuntime deployment but in opendatahub/odh-model-controller, we are creating `ServiceAccount` for user if its not already exists.

So if we follow kserve pattern then its a user responsiblity to delete old SA.

Reply:
> According to kserve/modelmesh-controller codebase, `ServiceAccount` should be maintained by the user.

I agree with you. As an independent component, it is user responsibility.

However, my understanding is that `odh-model-controller` is adding missing things so that ODH can be used as a platform. Because of this, I'd lean towards deleting automatically. It is worth to check with other people to confirm.